### PR TITLE
Nested charts with `d3-graph` component

### PR DESCRIPTION
**Pull request by taras**
PR for #19 

`d3-graph` can now be nested. The child components are rendered as contextual components with `tagName=''`

```hbs
{{#d3-graph as |d3|}}
  {{d3.graph (pipe ...)}}
  {{d3.graph (pipe ...)}}
{{/d3-graph}}
```

Making these contextual components has the unfortunate side effect of requiring Ember 2.3+. What are your thoughts on this?

/cc @ivanvanderbyl @zigahertz

### CODE REVIEW

**Comment by zigahertz** [comment]
How unfortunate is it to require v2.3+? Given that this library takes advantage of APIs that require 1.13+, and that the upgrade path from that version to 2.3 is usually pretty smooth, this restriction doesn't seem too bad to me. According to the [2016 survey](http://emberjs.com/ember-community-survey-2016/), a majority of developers are on 1.13 or higher. If we go with this, it would be important to mention that this library requires v2.3.

**Comment by taras** [comment]
@zigahertz `ember-paper` has examples using contextual components and not using contextual components. We could do the same, because the component will work with or without contextual components.

File: addon/templates/components/d3-graph.hbs [hunk 1]
```diff
@@ -1,6 +1,17 @@
-{{yield selection}}
 {{#if selection}}
-  {{#if graph}}
-    {{shhh (compute graph selection)}}
+  {{#if hasBlock}}
+    {{#if graph}}
```
Comment (zigahertz) [inline addon/templates/components/d3-graph.hbs]: is there a nicer way to do this than using a triply-nested `if` statement? maybe we can use `ember-truth-helpers` here?

**Comment by zigahertz** [comment]
can this PR also fix the fact that this test file is completely commented out:
https://github.com/LocusEnergy/ember-d3-helpers/blob/master/tests/integration/components/d3-graph-test.js
?

File: addon/templates/components/d3-graph.hbs (hunk 1 above)
Comment (taras) [inline addon/templates/components/d3-graph.hbs]: Good idea. Done.

**Comment by taras** [comment]
@zigahertz this PR adds tests for d3-graph

**Comment by zigahertz** [comment]
@taras looks great! thank you!

**Pull request merged**